File: mobile_apps/Delivera Swift iOS/DeliveraSwift/Core/DesignSystem/DeliveraTheme.swift
import SwiftUI

enum DeliveraTheme {
    static let orange = Color(red: 0.94, green: 0.31, blue: 0.10)
    static let ink = Color(red: 0.06, green: 0.06, blue: 0.07)
    static let muted = Color(red: 0.43, green: 0.42, blue: 0.40)
    static let panel = Color.white
    static let line = Color.black.opacity(0.065)
    static let gold = Color(red: 0.94, green: 0.73, blue: 0.36)

    static var appBackground: some View {
        LinearGradient(
            colors: [
                Color(red: 0.99, green: 0.98, blue: 0.95),
                Color(red: 0.96, green: 0.98, blue: 0.96),
                Color(red: 0.99, green: 0.96, blue: 0.93)
            ],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
    }
}

extension View {
    func cardShadow() -> some View {
        shadow(color: .black.opacity(0.075), radius: 18, x: 0, y: 10)
    }
}

struct DpointsGlyph: View {
    var size: CGFloat = 18

    var body: some View {
        RoundedRectangle(cornerRadius: size * 0.22, style: .continuous)
            .fill(DeliveraTheme.orange)
            .frame(width: size, height: size)
            .overlay {
                RoundedRectangle(cornerRadius: size * 0.1, style: .continuous)
                    .stroke(.white, lineWidth: max(1.5, size * 0.11))
                    .frame(width: size * 0.42, height: size * 0.42)
                    .rotationEffect(.degrees(45))
            }
    }
}

struct DpointsPriceBadge: View {
    let product: MenuProduct
    var valuePerKr: Double = 10
    var extrasTotal: Double = 0
    var quantity: Int = 1

    var body: some View {
        if product.rewardable == true {
            HStack(spacing: 5) {
                DpointsGlyph(size: 16)
                Text("\(product.dpointsUnitCost(valuePerKr: valuePerKr, extrasTotal: extrasTotal) * quantity) Dpoints")
                    .font(.system(size: 12, weight: .black))
                    .foregroundStyle(DeliveraTheme.orange)
            }
            .padding(.horizontal, 8)
            .frame(height: 26)
            .background(DeliveraTheme.orange.opacity(0.09), in: Capsule())
        }
    }
}

File: mobile_apps/Delivera Android/app/src/main/java/se/delivera/android/ui/theme/DeliveraTheme.kt
package se.delivera.android.ui.theme

import android.graphics.BlurMaskFilter
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.drawBehind
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Paint
import androidx.compose.ui.graphics.Shadow
import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import se.delivera.android.data.MenuProduct

/**
 * 1:1 port of DeliveraTheme.swift. Colours use the exact same float components
 * as the SwiftUI source so nothing shifts between platforms.
 */
object DeliveraTheme {
    val orange = Color(red = 0.94f, green = 0.31f, blue = 0.10f)
    val ink = Color(red = 0.06f, green = 0.06f, blue = 0.07f)
    val muted = Color(red = 0.43f, green = 0.42f, blue = 0.40f)
    val panel = Color.White
    val line = Color.Black.copy(alpha = 0.065f)
    val gold = Color(red = 0.94f, green = 0.73f, blue = 0.36f)

    // Deal cards. A light, clean blue identity (flat, no glow) shared by the hero
    // banner and the deal rail so everything feels like one system.
    val dealBlue = Color(red = 0.07f, green = 0.53f, blue = 0.96f)
    val dealBlueDeep = Color(red = 0.04f, green = 0.33f, blue = 0.85f)
    val dealBlueInk = Color(red = 0.03f, green = 0.24f, blue = 0.46f)
    val dealBlueChip = Color(red = 0.88f, green = 0.95f, blue = 1.0f)

    val dealBlueGradient: Brush
        get() = Brush.linearGradient(
            colors = listOf(dealBlue, dealBlueDeep),
            start = Offset(0f, 0f),
            end = Offset.Infinite
        )

    val appBackgroundBrush: Brush
        get() = Brush.linearGradient(
            colors = listOf(
                Color(red = 0.99f, green = 0.98f, blue = 0.95f),
                Color(red = 0.96f, green = 0.98f, blue = 0.96f),
                Color(red = 0.99f, green = 0.96f, blue = 0.93f)
            ),
            start = Offset(0f, 0f),
            end = Offset.Infinite
        )
}

/** shadow(color: .black.opacity(0.075), radius: 18, y: 10) equivalent. */
val cardShadowSpec = Shadow(
    color = Color.Black.copy(alpha = 0.075f),
    offset = Offset(0f, 10f),
    blurRadius = 18f
)

/**
 * Swift `cardShadow()`: black 7.5%, blur radius 18, offset y 10. Drawn with a
 * real blur (BlurMaskFilter) instead of Compose elevation, which cannot express
 * a soft, offset, low-alpha shadow.
 */
fun Modifier.cardShadow(cornerRadius: Dp = 20.dp): Modifier = drawBehind {
    val paint = Paint()
    val frameworkPaint = paint.asFrameworkPaint()
    frameworkPaint.color = android.graphics.Color.argb((0.075f * 255).toInt(), 0, 0, 0)
    frameworkPaint.maskFilter = BlurMaskFilter(18.dp.toPx(), BlurMaskFilter.Blur.NORMAL)
    val offsetY = 10.dp.toPx()
    val radius = cornerRadius.toPx()
    drawIntoCanvas { canvas ->
        canvas.drawRoundRect(
            left = 0f,
            top = offsetY,
            right = size.width,
            bottom = size.height + offsetY,
            radiusX = radius,
            radiusY = radius,
            paint = paint
        )
    }
}

/** DpointsGlyph from DeliveraTheme.swift: orange rounded square with a rotated white diamond. */
@Composable
fun DpointsGlyph(size: Dp = 18.dp) {
    Box(
        Modifier
            .size(size)
            .clip(RoundedCornerShape(size * 0.22f))
            .background(DeliveraTheme.orange),
        contentAlignment = Alignment.Center
    ) {
        Box(
            Modifier
                .size(size * 0.42f)
                .rotate(45f)
                .border(
                    width = maxOf(1.5.dp, size * 0.11f),
                    color = Color.White,
                    shape = RoundedCornerShape(size * 0.1f)
                )
        )
    }
}

/** DpointsPriceBadge from DeliveraTheme.swift: shown only when the product is rewardable. */
@Composable
fun DpointsPriceBadge(
    product: MenuProduct,
    valuePerKr: Double = 10.0,
    extrasTotal: Double = 0.0,
    quantity: Int = 1
) {
    if (product.rewardable != true) return
    Row(
        Modifier
            .height(26.dp)
            .clip(RoundedCornerShape(50))
            .background(DeliveraTheme.orange.copy(alpha = 0.09f))
            .padding(horizontal = 8.dp),
        horizontalArrangement = Arrangement.spacedBy(5.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        DpointsGlyph(size = 16.dp)
        Text(
            "${product.dpointsUnitCost(valuePerKr, extrasTotal) * quantity} Dpoints",
            fontSize = 12.sp,
            fontWeight = FontWeight.Black,
            color = DeliveraTheme.orange
        )
    }
}

private val DeliveraColorScheme = lightColorScheme(
    primary = DeliveraTheme.orange,
    background = Color(red = 0.99f, green = 0.98f, blue = 0.95f),
    surface = DeliveraTheme.panel,
    onPrimary = Color.White,
    onBackground = DeliveraTheme.ink,
    onSurface = DeliveraTheme.ink
)

@Composable
fun DeliveraAppTheme(content: @Composable () -> Unit) {
    MaterialTheme(colorScheme = DeliveraColorScheme) {
        Surface {
            Box(Modifier.fillMaxSize().background(DeliveraTheme.appBackgroundBrush)) {
                content()
            }
        }
    }
}

/** Continuous rounded rectangle used throughout the app. */
fun deliveraCorner(radius: Float) = RoundedCornerShape(radius)
